### PR TITLE
Adding user login audit logs

### PR DESCRIPTION
**Pull request by vivekhiwarkar**
### What does this PR change?

- Adding User login audit logs #61 

### Does the PR depend on any other PRs or Issues? If yes, please list them.

- No

### Checklist

I confirm, that I have...

- [X] Read and followed the contributing guide in `CONTRIBUTING.md`
- [ ] Added tests for this PR
- [ ] Formatted the code using `go fmt` (if applicable)
- [ ] Updated [documentation on the Paralus docs site](https://github.com/paralus/website/blob/main/docs) (if applicable)
- [ ] Updated `CHANGELOG.md`

### CODE REVIEW

File: proto/types/audit/audit.pb.go [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
-// 	protoc-gen-go v1.27.1
+// 	protoc-gen-go v1.28.1
```
Comment (niravparikh05) [inline proto/types/audit/audit.pb.go:3]: can we use protoc-gen-go v1.27.1 to maintain consistency and not upgrade as part of this feature.

File: proto/rpc/user/user.proto [hunk 2]
```diff
@@ -75,8 +75,18 @@ message UserForgotPasswordResponse { string recoveryLink = 1; }
 
 message UserDeleteApiKeysResponse {}
 message CliConfigRequest {}
+message AuditWebhook {}
```
Comment (niravparikh05) [inline proto/rpc/user/user.proto:78]: create two messages to follow buf standards, UserLoginAuditRequest and UserLoginAuditResponse

File: proto/rpc/user/group.pb.gw.go [hunk 3]
```diff
@@ -16,7 +16,7 @@ import (
 	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
 	"github.com/grpc-ecosystem/grpc-gateway/v2/utilities"
 	"github.com/paralus/paralus/proto/types/commonpb/v3"
-	"github.com/paralus/paralus/proto/types/userpb/v3"
+	userv3_0 "github.com/paralus/paralus/proto/types/userpb/v3"
```
Comment (niravparikh05) [inline proto/rpc/user/group.pb.gw.go:19]: looks like these are due to a different plugin version, let's avoid such changes as part of this feature. applicable to all other *pb files that are irrelavent to this feature.

File: pkg/service/user.go [hunk 4]
```diff
@@ -52,6 +52,8 @@ type UserService interface {
 	UpdateIdpUserGroupPolicy(context.Context, string, string, string) error
 	// Generate recovery link for users
 	ForgotPassword(context.Context, *userrpcv3.UserForgotPasswordRequest) (*userrpcv3.UserForgotPasswordResponse, error)
+	// Generate auditLog event
+	AuditLogWebhook(context.Context, *userrpcv3.AuditWebhook) (*userrpcv3.AuditWebhook, error)
```
Comment (niravparikh05) [inline pkg/service/user.go:56]: name it better at service layer - probably something like CreateLoginAuditLog

File: pkg/service/audit_utils.go [hunk 5]
```diff
@@ -480,3 +480,22 @@ func CreateLocationAuditEvent(ctx context.Context, al *zap.Logger, action string
 		_log.Warn("unable to create audit event", err)
 	}
 }
+
+func UserLoginAuditEvent(ctx context.Context, al *zap.Logger, action string, name string) {
```
Comment (niravparikh05) [inline pkg/service/audit_utils.go:484]:
```suggestion
func CreateUserLoginAuditEvent(ctx context.Context, al *zap.Logger, action string, name string) {
```

File: pkg/service/user.go [hunk 6]
```diff
@@ -1079,3 +1081,10 @@ func (s *userService) getUserLastLogin(ctx context.Context, userId uuid.UUID) (s
 	}
 	return lastLogin, nil
 }
+
+func (s *userService) AuditLogWebhook(ctx context.Context, req *userrpcv3.AuditWebhook) (*userrpcv3.AuditWebhook, error) {
+
+	UserLoginAuditEvent(ctx, s.al, "login", "username")
```
Comment (niravparikh05) [inline pkg/service/user.go:1087]: I suppose this was suppose to be the actual user name and not string "username"

File: proto/rpc/user/user.proto [hunk 7]
```diff
@@ -75,8 +75,18 @@ message UserForgotPasswordResponse { string recoveryLink = 1; }
...
+
+service UserService { 
+
+  rpc AuditLogWebhook(AuditWebhook)
+      returns (AuditWebhook) {
+    option (google.api.http) = {
+      post : "/userauditlog"
```
[the remark this answers is not in the document]
Comment (vivekhiwarkar) [reply proto/rpc/user/user.proto]: Changed to /auth/v3/user/auditlog in latest commit

File: pkg/service/user.go (hunk 4 above)
Comment (vivekhiwarkar) [reply pkg/service/user.go]: made the change

File: proto/types/audit/audit.pb.go (hunk 1 above)
Comment (vivekhiwarkar) [reply proto/types/audit/audit.pb.go:3]: Still figuring out on how to generate proto files with previous version, not marking this as resolved yet
@akshay196 @niravparikh05 do let me know if you have any context of this

File: proto/rpc/user/group.pb.gw.go (hunk 3 above)
Comment (vivekhiwarkar) [reply proto/rpc/user/group.pb.gw.go:19]: proto version issue, unresolved

File: pkg/service/audit_utils.go (hunk 5 above)
Comment (vivekhiwarkar) [reply pkg/service/audit_utils.go]: changed in latest commit

File: pkg/service/user.go (hunk 6 above)
Comment (vivekhiwarkar) [reply pkg/service/user.go]: This has to come in the request body from kratos, just added a placeholder string for now

**Comment by niravparikh05** [comment]
Closing this as we have a different [pull request](https://github.com/paralus/paralus/pull/111) created.

**Pull request closed without merging**